lib/Varien/File/Uploader.php

```diff
@@ -505,8 +505,8 @@ private function _setUploadFileId($fileId)
             preg_match("/^(.*?)\[(.*?)\]$/", $fileId, $file);
 
             if (count($file) > 0
-                && (count($file[0]) > 0)
-                && (count($file[1]) > 0)
+                && isset($file[0])
+                && isset($file[1])
```

Comment:
I did some tests on 
```php
preg_match("/^(.*?)\[(.*?)\]$/", $fileId, $file);`. 
```

`$fieldId` is int or null or 'abc', `$file` is `[]`.

`$fieldId` is '1[2]', `$file` is
```
 array(3) {
  [0] => string(4) "1[2]"
  [1] => string(1) "1"
  [2] => string(1) "2"
}
```

`$fieldId` is '[abc]', `$file` is
```
 array(3) {
  [0] => string(5) "[abc]"
  [1] => string(0) ""
  [2] => string(3) "abc"
}
```

`$fieldId` is 'abc[]', `$file` is
```
 array(3) {
  [0] => string(5) "abc[]"
  [1] => string(3) "abc"
  [2] => string(0) ""
}
```
$fieldIs is 'abc[xyz]', `$file` is
```
 array(3) {
  [0] => string(8) "abc[xyz]"
  [1] => string(3) "abc"
  [2] => string(3) "xyz"
}
```

The next statement
```php
array_shift($file);
```
removes the first element, now `$file` has the last 2 elements, both are string.

Given the above, we should look for non-empty `$file[1]` and `$file[2]`, `$file[0]` is irrelevant as it is removed.

```suggestion
            if (count($file) === 3 && $file[1] && $file[2]) {
```